### PR TITLE
Cut(eos_cli_config_gen): Remove the deprecated keys for port-channel-interfaces

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Remove the deprecated keys for port-channel-interfaces

## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Remove the deprecated keys for port-channel-interfaces

## How to test
Run molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/port-channel-interfaces.j2 [hunk 1]
```diff
@@ -120,10 +120,10 @@ interface {{ port_channel_interface.name }}
 {%     if port_channel_interface.flow_tracker.sampled is arista.avd.defined %}
    flow tracker sampled {{ port_channel_interface.flow_tracker.sampled }}
 {%     endif %}
-{%     if port_channel_interface.evpn_ethernet_segment.identifier | arista.avd.default(port_channel_interface.esi) is arista.avd.defined or port_channel_interface.evpn_ethernet_segment is arista.avd.defined %}
+{%     if port_channel_interface.evpn_ethernet_segment.identifier is arista.avd.defined or port_channel_interface.evpn_ethernet_segment is arista.avd.defined %}
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/port-channel-interfaces.j2]:
```suggestion
{%     if port_channel_interface.evpn_ethernet_segment is arista.avd.defined %}
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 2]
```diff
@@ -70,8 +70,8 @@ The following data model keys have been removed from `eos_cli_config_gen` in v5.
 | old key 4.2(management_api_gnmi) | new key(TODO) |
 | management_security.entropy_source | management_security.entropy_sources |
 | old key 6(name_server) | new key(TODO) |
-| old key 7.1(port_channel_interfaces) | new key(TODO) |
-| old key 7.2(port_channel_interfaces) | new key(TODO) |
+| port_channel_interfaces.esi | port_channel_interfaces.evpn_ethernet_segment.identifier |
```
Comment (laxmikantchintakindi) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md:81]: | port_channel_interfaces[].esi | port_channel_interfaces[].evpn_ethernet_segment.identifier |

Can we add like this.

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/port_channel_interfaces.schema.yml [hunk 3]
```diff
@@ -854,3 +836,23 @@ keys:
         eos_cli:
           type: str
           description: Multiline EOS CLI rendered directly on the port-channel interface in the final EOS configuration.
+        esi:
+          type: str
+          description: |
+            EVPN Ethernet Segment Identifier (Type 1 format).
+            If both "esi" and "evpn_ethernet_segment.identifier" are defined, the new variable takes precedence.
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/port_channel_interfaces.schema.yml]:
```suggestion
```

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/port_channel_interfaces.schema.yml [hunk 4]
```diff
@@ -854,3 +836,23 @@ keys:
...
+          deprecation:
+            removed: true
+            warning: true
+            new_key: evpn_ethernet_segment.identifier
+            remove_in_version: "5.0.0"
+        rt:
+          type: str
+          description: |
+            EVPN Route Target for ESI with format xx:xx:xx:xx:xx:xx.
+            If both "rt" and "evpn_ethernet_segment.route_target" are defined, the new variable takes precedence.
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/port_channel_interfaces.schema.yml]:
```suggestion
```

**Review by ClausHolbechArista — DISMISSED** [review]

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/port-channel-interfaces.j2 [hunk 5]
```diff
@@ -171,7 +171,7 @@
 {%         for evpn_es_po_interface in evpn_es_po_interfaces | arista.avd.natural_sort('name') %}
 {%             set esi = evpn_es_po_interface.evpn_ethernet_segment.identifier | arista.avd.default(evpn_es_po_interface.esi, "-") %}
```
Comment (laxmikantchintakindi) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/port-channel-interfaces.j2:172]:
```suggestion
{%             set esi = evpn_es_po_interface.evpn_ethernet_segment.identifier | arista.avd.default("-") %}
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md (hunk 2 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md:81]: yes

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md:81]: But you also need to add rt -> route_target

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md:81]: updated

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/port-channel-interfaces.j2 (hunk 1 above)
Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/port-channel-interfaces.j2]: fixed

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/port_channel_interfaces.schema.yml (hunk 3 above)
Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/port_channel_interfaces.schema.yml]: updated

File: python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/port_channel_interfaces.schema.yml (hunk 4 above)
Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/port_channel_interfaces.schema.yml]: updated

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/port-channel-interfaces.j2 (hunk 5 above)
Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/port-channel-interfaces.j2:172]: updated

**Review by laxmikantchintakindi — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Review by gmuloc — APPROVED** [review]

**Pull request merged**